codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/ServiceConfigGeneratorTest.kt

```diff
@@ -76,45 +85,100 @@ internal class ServiceConfigGeneratorTest {
+
                     ServiceConfig.BuilderStruct -> writable { rust("config_field: Option<u64>") }
                     ServiceConfig.BuilderImpl -> emptySection
                     ServiceConfig.BuilderBuild -> writable {
-                        rust("config_field: self.config_field.unwrap_or_default(),")
+                        if (runtimeMode.defaultToOrchestrator) {
+                            rustTemplate(
+                                """
+                                layer.store_or_unset(self.config_field.map(#{T}));
+                                """,
+                                "T" to configParamNewtype(
+                                    "config_field".toPascalCase(), RuntimeType.U64.toSymbol(),
+                                    codegenContext.runtimeConfig,
+                                ),
+                            )
+                        } else {
+                            rust("config_field: self.config_field.unwrap_or_default(),")
+                        }
                     }
+
                     else -> emptySection
                 }
             }
         }
-        val sut = ServiceConfigGenerator(listOf(ServiceCustomizer()))
-        val symbolProvider = testSymbolProvider("namespace empty".asSmithyModel())
+
+        val model = "namespace empty".asSmithyModel()
+        val smithyRuntimeMode = SmithyRuntimeMode.fromString(smithyRuntimeModeStr)
+        val codegenContext = testClientCodegenContext(model).withSmithyRuntimeMode(smithyRuntimeMode)
+        val sut = ServiceConfigGenerator(codegenContext, listOf(ServiceCustomizer(codegenContext)))
+        val symbolProvider = testSymbolProvider(model)
         val project = TestWorkspace.testProject(symbolProvider)
         project.withModule(ClientRustModule.Config) {
             sut.render(this)
-            unitTest(
-                "set_config_fields",
-                """
-                let mut builder = Config::builder();
-                builder.config_field = Some(99);
-                let config = builder.build();
-                assert_eq!(config.config_field, 99);
-                """,
-            )
+            if (smithyRuntimeMode.defaultToOrchestrator) {
+                unitTest(
+                    "set_config_fields",
+                    """
+                    let mut builder = Config::builder();
+                    builder.config_field = Some(99);
+                    let config = builder.build();
+                    assert_eq!(config.config_field(), 99);
+                    """,
+                )
+            } else {
+                unitTest(
+                    "set_config_fields",
+                    """
+                    let mut builder = Config::builder();
+                    builder.config_field = Some(99);
+                    let config = builder.build();
+                    assert_eq!(config.config_field, 99);
+                    """,
+                )
+            }
```

Comment:
Are these branches both the same or am I missing something?

Reply:
Admittedly, the downside of code duplication mentioned above is this: can be hard to spot where the difference is. The difference in this case is line 168 vs. line 178. One needs `config_field()` to access the field and the other can access it directly.